### PR TITLE
Finish functionality

### PR DESCRIPTION
**Pull request by David-Lanzz**
# Hello Reviewer, I have completed my task for today and below is a summary of my work
- No linter errors
- Correct use of git flow
- Completion of UI functionality
- Task completed

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,8 +3,12 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
   root "users#index"
-  resources :users,only:[:index,:show] do
-    resources :posts,only:[:index,:show]
+  resources :users,only:[:index,:show,:new,:create] do
+    resources :posts,only:[:index,:show,:new,:create] do
```
Comment (FrederickMih) [inline config/routes.rb:9]: - [ ] Hi @David-Lanzz, in the requirements of this project, we are told not to expose unnecessary routes. The ONLY permitted routes should be `#new` and `#create`. This is to minimize our memory usage :+1:

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    @post = Post.find(params[:post_id])
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params)
+    @comment.author_id = current_user.id
```
Comment (FrederickMih) [inline app/controllers/comments_controller.rb:10]: - [Optional] Just as you have done in the posts_controller, we could  equally do this: 

```
 @comment = current_user.comments.new(comment_params) 
 @comment.post_id = params[:post_id]
```

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    else
+      flash[:error] = @comment.errors.full_messages.join(', ')
+      render :new
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:title, :text)
```
Comment (FrederickMih) [inline app/controllers/comments_controller.rb]: - [ ] If you check the comments table once more, you'll discover that we need only one field to create a comment: The `text` field. We don't require the title of a comment.

File: app/views/posts/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <%= f.label :title %><br>
+    <%= f.text_field :title %>
+  </div>
+
+  <div>
+    <%= f.label :text %><br>
+    <%= f.text_area :text %>
+  </div>
+
+  <%= f.submit "Add post" %>
```
Comment (FrederickMih) [inline app/views/posts/new.html.erb:23]: - [ ] When I decided to create a new post by filling out the required fields, a post was not actually created when clicked on the "Add post" button. Please kindly make sure it is possible to create a post.

File: app/views/comments/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    <title>Document</title>
+</head>
+<body>
+    <%= form_with model: @comment, url: user_post_comments_path(user_id: current_user.id, post_id: @post.id), method: :post do |f| %>
+  <div>
+    <%= f.label :text %>
+    <%= f.text_area :text %>
+  </div>
+  <%= f.submit 'Add Comment' %>
+<% end %>
```
Comment (FrederickMih) [inline app/views/comments/new.html.erb]: - [ ] After making sure a post can be created, you'll equally kindly double-check if commenting and liking on a post is possible

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @David-Lanzz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Great job for making sure both posts and comments are created on behalf of the currently logged in user :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (rloterh) [reply config/routes.rb:9]: well resolved! ✔️

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (rloterh) [reply app/controllers/comments_controller.rb]: well resolved! ✔️

File: app/views/posts/new.html.erb (hunk 4 above)
Comment (rloterh) [reply app/views/posts/new.html.erb:23]: well resolved! ✔️

File: app/views/comments/new.html.erb (hunk 5 above)
Comment (rloterh) [reply app/views/comments/new.html.erb]: Needs improvement!

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    @post = Post.find(params[:post_id])
+  end
+
```
Comment (rloterh) [inline app/controllers/comments_controller.rb:6]: - [ ] A comment to a post cannot be created successfully as seen below. Let's ensure to improve upon this aspect.

[comment_issue.webm](https://github.com/David-Lanzz/lanzz-blog/assets/12745474/3304a2e8-fded-4944-b039-0391cc10fce7)

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Lanzz, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🌴  🏃 🌳 🏡 🏘️ 🌲 

## Highlights 👍  
- User can create post correctly!
- Like is working as intended!
- ReadMe looks good!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @David-Lanzz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**